qa/tasks/cephfs/test_exports.py

```diff
@@ -175,3 +175,47 @@ def test_session_race(self):
         # Check if rank1 changed (standby tookover?)
         new_rank1 = self.fs.get_rank(rank=1)
         self.assertEqual(rank1['gid'], new_rank1['gid'])
+
+    def test_ephememeral_pin(self):
+
+
+        self.fs.set_max_mds(3)
+        self.fs.wait_for_daemons()
+
+        status = self.fs.status()
+
+        self.mount_a.run_shell(["mkdir", "-p", "a"])
+        self._wait_subtrees(status, 0, [])
+
+        for i in range(0,100):
+          self.mount_a.run_shell(["mkdir", "-p", "a/" + i + "/d"])
+          
+        self._wait_subtrees(status, 0, [])
+
+        self.mount_b.setfattr(["a", "ceph.dir.distributed.pin", "1"])
+
+        self._wait_distributed_subtrees([status, 0, 100)]
+
+        rank0_distributed_subtree_ratio = len(self.get_distributed_auth_subtrees(status, 0)/len(self.get_auth_subtrees(status, 0))
+        self.assertGreaterEqual(rank0_distributed_subtree_ratio, 0.2)
+
+        rank1_distributed_subtree_ratio = len(self.get_distributed_auth_subtrees(status, 1)/len(self.get_auth_subtrees(status, 1))
+        self.assertGreaterEqual(rank1_distributed_subtree_ratio, 0.2)
+
+        rank2_distributed_subtree_ratio = len(self.get_distributed_auth_subtrees(status, 2)/len(self.get_auth_subtrees(status, 2))
+        self.assertGreaterEqual(rank2_distributed_subtree_ratio, 0.2)
```

Comment:
Test should stop here.

Add another test for growing max_mds. Add a third test for shrinking max_mds.